core/io/linux.rs

```diff
@@ -83,11 +84,14 @@ impl WrappedIOUring {
     fn submit_entry(&mut self, entry: &io_uring::squeue::Entry, c: Rc<Completion>) {
         log::trace!("submit_entry({:?})", entry);
         self.pending.insert(entry.get_user_data(), c);
-        unsafe {
-            self.ring
-                .submission()
-                .push(entry)
-                .expect("submission queue is full");
+        loop {
+            let res = unsafe { self.ring.submission().push(entry) };
+            match res {
+                Ok(_) => break,
+                Err(_) => {
+                    let _ = self.wait_for_completion();
+                }
```

Comment:
I'm not a fan of this, we are ignoring the error and we are waiting for completion for everything. This feels kinda wrong honestly because that would mean that if you try to write a big blob it will for the connection to wait instead of returning control to it as we would expect from limbo.

In my opinion submit_entry should return submission is full and therefore there should be a state machine in write path that can resume long writes on next iteration.

cc: @penberg